pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -37,6 +38,16 @@ import (
 	"k8s.io/kubernetes/pkg/util/hash"
 )
 
+// SemanticIgnoreResourceVersion does semantic deep equality checks for objects
+// but excludes ResourceVersion of ObjectReference.
+var SemanticIgnoreResourceVersion = conversion.EqualitiesOrDie(
```

Comment:
a few thoughts:
* something alone these lines might be ok, but we want to be coherent all the places we're checking endpoints / endpointslices / subsets
* we should limit use of this as a general purpose comparison... if we only expect this to be used to compare Subsets, or Endpoints, or EndpointSlices (and only for certain purposes), can we limit use to those types by exporting comparison functions just for those with appropriate doc?
* do `EndpointsEqualBeyondHash` and `objectRefPtrChanged` need to adjust here as well?

Reply:
1. Added `EndpointSubsetsEqualIgnoreResourceVersion` for Endpoints API and updated `EndpointsEqualBeyondHash` for EndpointSlice API to make their behavior consistent
2.  Made the variable private and used it only in `EndpointSubsetsEqualIgnoreResourceVersion` and `EndpointsEqualBeyondHash`
3. Updated `EndpointsEqualBeyondHash` and removed `objectRefPtrChanged` which is no longer needed